src/error.rs

```diff
@@ -122,12 +130,20 @@ impl From<UserError> for Error {
 
 impl fmt::Display for Error {
     fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
-        use self::Kind::*;
-
         match self.kind {
-            Proto(ref reason) => write!(fmt, "protocol error: {}", reason),
-            User(ref e) => write!(fmt, "user error: {}", e),
-            Io(ref e) => fmt::Display::fmt(e, fmt),
+            Kind::Reset(_, reason, initiator) => {
+                write!(fmt, "stream reset by {}: {}", initiator, reason)
+            }
+            Kind::GoAway(ref debug_data, reason, initiator) => {
+                write!(fmt, "go away from {}: {}", initiator, reason)?;
```

Comment:
I'm not sure my thought is useful, but I figured I'd list it:

I kinda like `"connection error detected: flow control error"` and `"connection error received (goaway): flow control error"`. I suppose it might be less clear, but at the same time it feels like it makes more sense from the perspective of a user who doesn't know what a "reset" or "go away" is... 🤷